tests/transformers/test_tokenizer.py

```diff
@@ -0,0 +1,147 @@
+import shutil
+import unittest
+from multiprocessing import Process
+from tempfile import TemporaryDirectory
+from parameterized import parameterized
+
+from paddle import nn
+from paddlenlp.transformers.model_utils import PretrainedModel, MODEL_HOME
+from paddlenlp.transformers.tokenizer_utils import PretrainedTokenizer
+
+from paddlenlp.transformers import BertTokenizer, BasicTokenizer, WordpieceTokenizer
+
+from paddlenlp.transformers.bert.modeling import BertForPretraining
+from paddlenlp.transformers.gpt.modeling import GPTForPretraining
+from paddlenlp.transformers.tinybert.modeling import TinyBertForPretraining
+
+from paddlenlp.transformers.bert.tokenizer import BertTokenizer
+from paddlenlp.transformers.gpt.tokenizer import GPTTokenizer, GPTChineseTokenizer
+from paddlenlp.transformers.tinybert.tokenizer import TinyBertTokenizer
+
+from tests.common_test import CpuCommonTest, CommonTest
+from tests.util import slow, assert_raises
+
+
+class FakePretrainedModel(PretrainedModel):
+
+    def __init__(self):
+        super().__init__()
+        self.linear = nn.Linear(2, 2)
+
+    def forward(self, *args, **kwargs):
+        pass
+
+
+def get_pretrained_models_params() -> List[Tuple[str, Type[PretrainedModel]]]:
+    """get all of pretrained model names in some PretrainedModels
+
+    Returns:
+        List[Tuple[str, Type[PretrainedModel]]]: the parameters of unit test method
+    """
+    model_types: List[PretrainedModel] = [
+        BertForPretraining, GPTForPretraining, TinyBertForPretraining
+    ]
+    name_class_tuples: List[Tuple[str, Type[PretrainedModel]]] = []
+    for ModelType in model_types:
+        for model_name in ModelType.pretrained_resource_files_map.get(
+                'model_state', {}).keys():
+            name_class_tuples.append([model_name, ModelType])
+    return name_class_tuples
+
+
+def get_pretrained_tokenzier_params(
+) -> List[Tuple[str, Type[PretrainedTokenizer]]]:
+    """get all of pretrained tokenzier names in some PretrainedTokenzier
+
+    Returns:
+        List[Tuple[str, Type[PretrainedTokenzier]]]: the parameters of unit test method
+    """
+    tokenizer_types: List[PretrainedTokenizer] = [
+        BertTokenizer, GPTTokenizer, GPTChineseTokenizer, TinyBertTokenizer
+    ]
+    name_class_params: List[Tuple[str, Type[PretrainedTokenizer]]] = []
+    for TokenizerType in tokenizer_types:
+        for model_name in TokenizerType.pretrained_resource_files_map.get(
+                'vocab_file', {}).keys():
+            name_class_params.append([model_name, TokenizerType])
+    return name_class_params
+
+
+class TestPretrainedFromPretrained(CpuCommonTest):
+    """module for test pretrained model"""
+
+    def setUp(self):
+        self.temp_dir = TemporaryDirectory()
+        model = FakePretrainedModel()
+        model.save_pretrained(self.temp_dir.name)
+
+    def do_pretrained_in_process(self):
+        FakePretrainedModel.from_pretrained(self.temp_dir.name)
```

Comment:
Need to test multi-processing of tokenizer since `PretrainedModel` use check before saving while `PretrainedTokenizer` not.

Reply:
Yes, there are also some improvements to do:
- [ ] test `PretrainedTokenizer` under multi-processing
- [ ] change  multi-processing under multi-stage k(0<k<=8)  processors which will be more stable for the testing. 

All of above improvements will be done in the next few days.